### DISCUSSION TITLE
Is there a way to search through a result set with different keys?

### DISCUSSION THREAD

**Discussion by elioncho** [TarekRaafat/autoComplete.js#191] [General]
I am querying an API that returns an array of objects. These objects have different structures:

{ "name": "a_name" ... }
{ "title": "a_title" ... }

Can I search by name and title? Or do the result set has to have the same keys?

**Comment by TarekRaafat** [comment]
Hello @elioncho,

All you have to do is adding all the keys you want to include for the search in the array of `data.keys` method. 
Check the API [Config](https://tarekraafat.github.io/autoComplete.js/#/?id=api-configuration) for clarification.

Try it and let me know how it went.

**Reply by elioncho** [reply]
Thanks for the reply. Just to clarify, the objects in the array can have different keys. So a key could exist in one object but not the other one. I get the following error:

`Uncaught (in promise) TypeError: Cannot read property 'toString' of undefined`

**Reply by TarekRaafat** [reply]
My bad, a quick solution could be easily normalizing the input data inside the `data.src` method by adding the same keys with a value of `null` or any other value you would like since they are missing to have a unified data object.
